tasmota/tasmota_xsns_sensor/xsns_91_vindriktning.ino

```diff
@@ -96,9 +107,37 @@ bool VindriktningReadData(void) {
 
 /*********************************************************************************************/
 
+void VindriktningSendRequest() { // command for PM1006 sensor - request PM values
+
+  uint8_t tx_request[] = {0x11, 0x02, 0x0B, 0x01, 0xE1}; // see https://threadreaderapp.com/thread/1415291684569632768.html
+
+  if (PinUsed(GPIO_VINDRIKTNING_TX)) {
+    VindriktningSerial->write(tx_request, sizeof(tx_request));
+  } else {
+    AddLog(LOG_LEVEL_ERROR, PSTR("VDN: Can not send data to VINDRIKTNING PM1006, VINDRIKTNING Tx pin not configured"));
+  }
+
+  AddLog(LOG_LEVEL_DEBUG_MORE, PSTR("VDN: Tx %*_H"), sizeof(tx_request), tx_request);
+}
+
+/*********************************************************************************************/
+
 void VindriktningSecond(void) {                // Every second
+
+  if (PinUsed(GPIO_VINDRIKTNING_TX)) { // commands for PM1006 has to be sent by Tasmota ESPxx - no original VINDRIKTNING microcontroller available
+    if (PinUsed(GPIO_VINDRIKTNING_FAN)) {
+      if (TasmotaGlobal.tele_period == Settings->tele_period - VINDRIKTNING_FAN_SECONDS_BEFORE_TELEPERIOD) { // VINDRIKTNING PM1006 fan control - switch fan on VINDRIKTNING_FAN_SECONDS_BEFORE_TELEPERIOD seconds before tele_period measurement
+        digitalWrite(Pin(GPIO_VINDRIKTNING_FAN), HIGH);
+      } else if (TasmotaGlobal.tele_period == 1 && Settings->tele_period > (VINDRIKTNING_FAN_SECONDS_BEFORE_TELEPERIOD + 1)) { // VINDRIKTNING PM1006 fan control - switch fan off 1 second after tele_period measurement, for short tele_periods do not switch fan off at all
+        digitalWrite(Pin(GPIO_VINDRIKTNING_FAN), LOW);
+      }
+    } // if (PinUsed(GPIO_VINDRIKTNING_FAN))
+
+    VindriktningSendRequest(); // request fresh value from VINDRIKTNING PM1006 every second, to get fresh values on web immediately (as values from other sensors)
```

Comment:
Are you sure it is a good idea to request data every second? I don't know if PM1006 can answer so fast but even then, if it happens after fan has been switched off, the sensor readings might skew until fan has been run again at least one minute.

I would show continuous reading in web only when fan is running all time. Otherwise just mimic VINDRIKTNING behavior and update web reading also only at every teleperiod. What would you think about that?

Reply:
Hello Cougar, thank you for reviewing the code.

Original microcontroller in IKEA VINDRIKTNING is requesting PM1006 sensor for a new value each second, Tasmota before my chages use ESP to only read sensor values (which are available each second).
For cases of original IKEA board fully replaced by an ESP board (e.g. LaskaKit ESP-VINDRIKTNING), Tasmota is driving PM1006 sensor itself (this modification) and I decided not to change this original behaviour.

Based on arendst's comment above, I learned purpose of teleperiod and designed fan control to provide enough time advance before measurement (and values export) takes place - for fresh air to be delivered into PM1006 sensor to provide accurate results.
Yes, you are right, web UI values with stopped fan may not be accurate for longer teleperiods, so it makes sense to stop updating them. I can implement this change, of course.

Additional note: As IKEA VNDRIKTNING device is well suitable to fit also an HT sensor (e.g. SCD40/41 supported by Tasmota), I tuned timing of PM1006 fan switching for different teleperiods (by lot of measurements and comparing reported values to a reference temp meter), because VINDRIKTNING housing is inside heated by own ESP dissipated heat (esp. for ESP32). The tiny PM1006 fan needs ~1 minute to decrease internal temperature to the ambient level. This value is a parameter in source code and could be potentially turned into a runtime parameter.

Regards,
Jan